server/src/test/kotlin/suwayomi/tachidesk/test/ApplicationTest.kt

```diff
@@ -138,12 +139,22 @@ open class ApplicationTest {
             System.setProperty("org.eclipse.jetty.util.log.class", "org.eclipse.jetty.util.log.StdErrLog")
             System.setProperty("org.eclipse.jetty.LEVEL", "OFF")
 
-            // socks proxy settings
-            if (serverConfig.socksProxyEnabled) {
-                System.getProperties()["socksProxyHost"] = serverConfig.socksProxyHost
-                System.getProperties()["socksProxyPort"] = serverConfig.socksProxyPort
-                logger.info("Socks Proxy is enabled to ${serverConfig.socksProxyHost}:${serverConfig.socksProxyPort}")
-            }
+            serverConfig.subscribeTo(
+                combine(
+                    serverConfig.socksProxyEnabled,
+                    serverConfig.socksProxyHost,
+                    serverConfig.socksProxyPort
+                ) { proxyEnabled, proxyHost, proxyPort ->
+                    Triple(proxyEnabled, proxyHost, proxyPort)
+                },
+                { (proxyEnabled, proxyHost, proxyPort) ->
+                    if (proxyEnabled) {
+                        System.getProperties()["socksProxyHost"] = proxyHost
+                        System.getProperties()["socksProxyPort"] = proxyPort
+                        logger.info("Socks Proxy is enabled to $proxyHost:$proxyPort")
+                    }
+                }
+            )
```

Comment:
Don't think this is needed in the tests.

Reply:
reverted